validator/src/main/java/com/amazon/aoc/validators/ValidatorFactory.java

```diff
@@ -43,7 +44,7 @@ public IValidator launchValidator(ValidationConfig validationConfig) throws Exce
     FileConfig expectedData = null;
     switch (validationConfig.getValidationType()) {
       case "trace":
-        validator = new TraceValidator();
+        validator = new TraceValidator(new XRayService(context.getRegion()), 2, 5);
```

Comment:
nit: Curious why we would move this out of the `init` function and into the constructor. Other two validators have the default values as `DEFAULT_MAX_RETRY_COUNT` like [here](https://github.com/aws-observability/aws-application-signals-test-framework/blob/17c094f770bab2bcb74f858264adfa75b164e3f9/validator/src/main/java/com/amazon/aoc/validators/CWLogValidator.java#L40). Would be best to keep it consistent if possible

Reply:
This small refactoring is mainly for unit test. 

Previously the corresponding CloudWatch service(metric, log, trace) is initialised directly within the init method instead of passing it through a parameter which made the unit test hard to mock this service. Hence, these validators are using a setter method for the CloudWatch service just for unit test to be able to overwrite it with a mock. 

Creating a new object in a class is not the best approach as it not just hard to test, but also introduce high coupling. 

My plan is to refactor MetricValidator and LogValidator in the next few PRs to also decouple CloudWatch service from it and get rid of these setter methods.

Same for the retry count, they are defined within the validator. Passing it through constructor could let unit test fail fast instead of retrying til the max limit.